src/ekat/io/ekat_yaml.cpp

```diff
@@ -303,47 +304,88 @@ std::string write_param<char> (const char& t) {
   return (t ? "true" : "false");
 }
 
-void write_parameter_list (const ParameterList& params, std::ostream& out, int indent) {
-  std::string tab(indent,' ');
-
-  // Helper lambda, to try all possible value types
+void write_parameter_list (const ParameterList& params, YAML::Emitter& out) {
+  // Helper lambda, to try all possible value types, return whether we found the type
   auto try_values = [&](const std::string& pname) -> bool {
-    bool found = false;
-    TypeListFor<values_t>([&](auto t) ->bool {
-      using vtype = decltype(t);
-      if (params.isType<vtype>(pname)) {
-        auto v = params.get<vtype>(pname);
-        out << write_param(v) << '\n';
-        found = true;
-      }
-      return found;
-    });
-    return found;
+    if (params.isType<int>(pname)) {
```

Comment:
The above syntax with `TypeListFor` seemed overly complicated, especially considering we know a priori the types, and there are only 4 of them...